### DISCUSSION TITLE
GraphQL: Return newest draft if one exists; fallback to published entry

### DISCUSSION THREAD

**Discussion by weotch** [craftcms/cms#8969] [Ideas]
I'd like an addition option for the `drafts` query argument so that it prefers drafts but it isn't a hard requirement.  This would be similar to how [Contentful's Content Preview API](https://www.contentful.com/developers/docs/references/content-preview-api/) works where a query for a given entry returns the draft state, if it exists, else it returns the published entry.  

My use case is for the staging version of a website where we'd like our users to be able to view their most draft content without requiring a preview token.

My proposal is that `drafts` could have the following values:

- `false` - No drafts returned (current behavior)
- `true` - Only drafts returned (current behavior)
- `"newest"` - Most recent drafts returned if they exist

We can workaround by using two `entry()` queries; this would just be more elegant.

**Comment by brandonkelly** [comment]
This would need to be possible with normal element queries first.

Related ideas: #2116, #6987
